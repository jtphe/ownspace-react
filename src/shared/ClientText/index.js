/* eslint-disable global-require */
import React from 'react';
import { Text, StyleSheet } from 'react-native';
import { useFonts } from '@use-expo/font';

/**
 * The custom TextClient component
 * @param {object} style - Style of the Text component
 * @param {string} ellipsizeMode - Defines how text will be truncated
 * @param {boolean} selectable - Lets the user select text, to use the native copy and paste functionality
 * @param {string} children - The string of the Text component
 */
const TextClient = props => {
  useFonts({
    DejaVuSans: require('../../../assets/fonts/DejaVuSans.ttf'),
    DejaVuSansBold: require('../../../assets/fonts/DejaVuSans-Bold.ttf')
  });

  const { style, ellipsizeMode, selectable, children } = props;

  if (style && style.fontWeight === 'bold') {
    /**
     * Render the TextClient component
     * @returns {React.Component} - TextClient component
     */
    return (
      <Text
        style={[style, styles.fontBold]}
        ellipsizeMode={ellipsizeMode}
        selectable={selectable}
      >
        {children}
      </Text>
    );
  }

  /**
   * Render the TextClient component
   * @returns {React.Component} - TextClient component
   */
  return (
    <Text
      style={[style, styles.fontNormal]}
      ellipsizeMode={ellipsizeMode}
      selectable={selectable}
    >
      {children}
    </Text>
  );
};

/**
 * Styles of TextClient component
 */
const styles = StyleSheet.create({
  fontNormal: {
    fontFamily: 'DejaVuSans'
  },
  fontBold: {
    fontFamily: 'DejaVuSansBold'
  }
});

export default TextClient;
