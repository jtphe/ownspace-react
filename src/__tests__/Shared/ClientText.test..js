import { Text } from 'react-native';
import React from 'react';
import ClientText from '@shared/ClientText';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('ClientText component tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ClientText />);
  });

  it('ClientText snapshot', async () => {
    let component;
    await renderer.act(async () => {
      component = renderer.create(<ClientText />);
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Check client's font", () => {
    const clientText = wrapper.find(Text).get(0);
    expect(clientText.props.style[1].fontFamily).toEqual('DejaVuSans');
  });

  it('Text selectable', () => {
    const clientText = wrapper.find(Text).get(0);
    expect(clientText.props.selectable).toEqual(true);
  });
});

Enzyme.configure({ adapter: new Adapter() });
