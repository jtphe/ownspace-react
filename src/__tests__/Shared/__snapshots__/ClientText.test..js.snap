// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientText component tests ClientText snapshot 1`] = `
<Text
  ellipsizeMode="tail"
  selectable={true}
  style={
    Array [
      null,
      Object {
        "fontFamily": "DejaVuSans",
      },
    ]
  }
/>
`;
